### DISCUSSION TITLE
Alignment Issue

### DISCUSSION THREAD

**Discussion by pratikshap31** [fabricjs/fabric.js#8328] [Q&A]
Hello,

Version : 1.6.0-rc.1
JS: Fabric.js
Issue: Tosvg coordinates value not correct
TestCase:
site: https://text.flyerco.dev/
Editor URL: https://text.flyerco.dev/flyers/3238/edit
Email Id: [text@mailinator.com](mailto:text@mailinator.com)
Password: text@123
Snap Video:

https://www.loom.com/share/b3ebff187fa14c45bdce3d1e6a1bb008
editor snap: https://www.awesomescreenshot.com/image/32390859?key=02a66fe1d846ef7a1a6b9158fb635cad
SVG snap: https://www.awesomescreenshot.com/image/32391978?key=f168d78065042113ae973f0dfc50a00a
PDF snap: https://www.awesomescreenshot.com/image/32390792?key=d950ff35e614901d3a136563236e301a

Description:
I am facing an issue while converting JSON objects to toSvg using fabric JS. It occurs in some cases with the right text alignment.
I don't understand the calculation of the SVG converter.
Does anyone know how to solve it?
<tspan xml:space="preserve" x="152.8" y="319.92" fill=" #000000">ius.2</tspan>
I have checked SVG data and the issue was found in X value when text right alignment for some font size.
This issue occurs when the last line of the paragraph has 1-2 words.
P.S: In my case, it occurs for some font sizes only. for example, 18,22 etc.

Let me know if any code, version, or anything needs to solve it

**Comment by ShaMan123** [comment]
VERSION

**Reply by ShaMan123** [reply]
4 major behind

**Reply by pratikshap31** [reply]
1.6.0-RC.1

I have tried to update to the latest version but cannot upgrade it as my site uses some deprecated methods and has minified mapped Javascript. 

Can you please help me here?
I have been solving this issue for the last 3 months.

**Reply by ShaMan123** [reply]
Use the changelog and the release notes to migrate.
http://fabricjs.com/articles/

v3 is considered dead, v4 is ending its life, v5 has entered LTS so v1?

**Reply by ShaMan123** [reply]
It is a waste of time not only for us, but for you as well.
Consider the bugs that were solved meanwhile.

**Reply by pratikshap31** [reply]
[editor.zip](https://github.com/fabricjs/fabric.js/files/9695889/editor.zip)
I have attached js file here. there are 15-17 js files mapped. how can I upgrade the version of fabric js?
